scriptmodules/supplementary/launchingimages.sh

```diff
@@ -0,0 +1,413 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="launchingimages"
+rp_module_desc="Generate runcommand launching images based on emulationstation themes."
+rp_module_help="A runcommand launching image is displayed while loading a game, with this tool you can automatically create some cool images based on a chosen emulationstation theme you have on your system."
+rp_module_section="exp"
+rp_module_flags="noinstclean"
+
+function depends_launchingimages() {
+    local depends=(imagemagick librsvg2-bin)
+    if isPlatform "x11"; then
+        getDepends feh
+    else
+        getDepends fbi
+    fi
+    getDepends "${depends[@]}"
+}
+
+function sources_launchingimages() {
+    gitPullOrClone "$md_inst" "https://github.com/meleu/generate-launching-images.git"
+}
+
+function install_launchingimages() {
+    cd "$md_inst"
+    chmod a+x generate-launching-images.sh
+}
+
+function _show_images_launchingimages() {
+    [[ -f "$1" ]] || return 1
+
+    local image="$1"
+    local timeout=5
+    local is_list=0
+
+    [[ "$(file -i $1)" =~ text/plain ]] && is_list=1
+
+    if isPlatform "x11"; then
+        feh \
+            --cycle-once \
+            --hide-pointer \
+            --fullscreen \
+            --auto-zoom \
+            --no-menus \
+            --slideshow-delay $timeout \
+            --quiet \
+            $([[ "$is_list" -eq 1 ]] && echo --filelist) \
+            "$image"
+    else
+        fbi \
+            --once \
+            --timeout "$timeout" \
+            --noverbose \
+            --autozoom \
+            $([[ "$is_list" -eq 1 ]] && echo --list) \
+            "$image" </dev/tty &>/dev/null
+    fi
+}
+
+function _dialog_menu_launchingimages() {
+    local text="$1"
+    shift
+    [[ -z "$@" ]] && return 1
+
+    # when there's only one item for the menu, the 'dialog --menu' expect
+    # to receive a "tag" and an "item" even if using '--no-item'. It can be
+    # an edge case, but this function shouldn't crash when it happens.
+    if [[ "$#" -eq 1 ]]; then
+        _dialog_yesno_launchingimages "$text: $1"
+        return
+    fi
+
+    dialog --backtitle "$__backtitle" --no-items --menu "$text" 22 86 16 "$@" 2>&1 >/dev/tty
+}
+
+function _set_theme_launchingimages() {
+    _dialog_menu_launchingimages \
+        "List of available themes" \
+        $("$md_inst/generate-launching-images.sh" --list-themes) \
+    || echo "\$theme"
```

Comment:
I would have thought the last line should be indented too, if splitting onto lots of lines.